### PR TITLE
Add background syncing test

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add background syncing test

## Links to any relevant issues

#387 

## Type of change

- Enhancement

## How the change has been tested

Running it locally with a private tangle

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: sdk/tests/wallet/syncing.rs [hunk 1]
```diff
@@ -148,3 +148,40 @@ async fn sync_only_most_basic_outputs() -> Result<()> {
...
+        let balance = account.balance().await?;
+        if balance.base_coin().available() > 0 {
+            break;
+        }
+    }
+
+    // Balance should be != 0 without calling account.sync()
+    let balance = account.balance().await?;
+    if balance.base_coin().available() == 0 {
+        panic!("Faucet no longer wants to hand over coins or background syncing failed");
```
Comment (kwek20) [inline sdk/tests/wallet/syncing.rs:182]: this should probably reflect what were testing, not what we encounter

File: sdk/tests/wallet/syncing.rs [hunk 2]
```diff
@@ -148,3 +148,40 @@ async fn sync_only_most_basic_outputs() -> Result<()> {
 
     tear_down(storage_path)
 }
+
+#[tokio::test]
+#[cfg(feature = "storage")]
+async fn background_syncing() -> Result<()> {
```
Comment (thibault-martinez) [inline sdk/tests/wallet/syncing.rs:155]: Should probably be an ignored test?

File: sdk/tests/wallet/syncing.rs (hunk 1 above)
Comment (Thoralf-M) [reply sdk/tests/wallet/syncing.rs:182]: Why not both? I think this way it's easier to debug if it fails

File: sdk/tests/wallet/syncing.rs (hunk 2 above)
Comment (Thoralf-M) [reply sdk/tests/wallet/syncing.rs:155]: Yes, added

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Sad clippy noises

File: sdk/tests/wallet/syncing.rs (hunk 1 above)
Comment (kwek20) [reply sdk/tests/wallet/syncing.rs:182]: i suppose, feels off for a traditional "test" but whatever :P

**Review by kwek20 — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**